### ISSUE TITLE
Time off by one hour

### ISSUE THREAD

**Issue by 0323pin** [grtcdr/citron#3]
Hi,
So, I've settled on `citron` with `lxqt-notificationd` as daemon but, the same happens with `dunst`.
The date readout is off by one hour, see screenshot and compare the time in the shell and the `citron` output.

![2021-12-07-114335_1366x768_scrot](https://user-images.githubusercontent.com/90570748/145016477-4c61a242-6a5c-4b64-b89d-4312ba3ad09d.png)

**Comment by grtcdr** [comment]
I think you have the same issue as me, where UTC is off by one hour, and local time is correct.

I'll add some command-line options to specify your time zone, or maybe read it automatically from a system file.

**Comment by 0323pin** [comment]
So, I'm not alone 😆

**Comment by grtcdr** [comment]
Nope.

But I'm not sure what the best solution is, in the `unstable` branch I've switched from UTC to using the local time, which does fix the issue, but I'm not sure if this is workaround is universal.

**Comment by grtcdr** [comment]
My hardware clock is off by one, and when I fixed it, it seemed to have reset itself when I booted into Linux. What the heck?

**Comment by 0323pin** [comment]
My hardware is set to local time (Stockholm, Sweden), date on the shell reports the same time because I've set it up this way.
Even though, the NetBSD guide states the following

> NetBSD, like all Unix systems, uses a system clock based on UTC (Coordinated Universal Time) and this is what you should set your system clock to. If you want to keep the system clock set to the local time (because, for example, you have a dual boot system with Windows installed), you must notify NetBSD, adding rtclocaltime=YES to /etc/rc.conf:
`# echo rtclocaltime=YES >> /etc/rc.conf`

**Comment by 0323pin** [comment]
Now, a couple of reboots later ...
If I remove `rtclocaltime=YES` from `/etc/rc.conf`, `citron` now displays the correct time but, my shell is off by one hour in the opposite direction.

**Comment by grtcdr** [comment]
I don't use Windows, but my system is behaving abnormally, just like the guide you mentioned. The hardware clock shows a correct output when running `hwclock --show` but timedatectl shows an incorrect RTC time.

What's going on here?

**Comment by 0323pin** [comment]
@grtcdr I don't have Windows either, and I don't know what's going on either :(

Please read section 5.9, http://www.netbsd.org/docs/guide/en/chap-boot.html#chap-boot-system-time

I have the symlink to the correct time zone, I have the bios clock set to local time and wait ... another reboot shoul do it if I'm correct.

**Comment by grtcdr** [comment]
> https://www.netbsd.org/docs/guide/en/chap-boot.html#chap-boot-system-time

This confuses me even more, I have to look into the details of UTC and what differs it from local time to be able to implement a correct report of time no matter the timezone.

**Comment by 0323pin** [comment]
So, I've removed `rtclocaltime=YES` from `/etc/rc.conf`, set my BIOS clock to UTC and kept the symlink to the correct time zone. Now my shell displays the correct time without the rtclocaltime work around 😄 
But `citron` is off by -1h again :(

So far I've only managed to get one of them correct. When `citron` is correct, my system is off by +1h

**Comment by grtcdr** [comment]
I'd say trust your system and not `citron`. I'll finalize `unstable` and release it, it might be better to use `--utc` than to get incorrect data, at least in the meantime.

**Comment by grtcdr** [comment]
Heads up, `--date-time` is now `--date`, but the shorthand flag hasn't changed.

**Comment by 0323pin** [comment]
> Heads up, `--date-time` is now `--date`, but the shorthand flag hasn't changed.

Saw that, thanks.

**Comment by grtcdr** [comment]
I might have found a way to automatically get the correct time by following /etc/localtime, I just need you to confirm what the symlink actually points to, on my machine it points to `../usr/share/zoneinfo/<region>/<country>`. Does yours show the same?

**Comment by 0323pin** [comment]
Yes. I can't confirm the two periods, as I've just switch-off my laptop but, that's the symlink which should be set by the installer.

Btw, nice screenshot and thanks for the feedback by e-mail ;)

**Comment by grtcdr** [comment]
> Yes. I can't confirm the two periods, as I've just switch-off my laptop but, that's the symlink which should be set by the installer.

Hmmmph, I really hope it's two periods on all of *nix.
 
> Btw, nice screenshot

Thank you :heart: 

> and thanks for the feedback by e-mail ;)

Anytime :)

**Comment by 0323pin** [comment]
> Hmmmph, I really hope it's two periods on all of *nix.

I can check this tomorrow of you want.
I just need to shell-out from `nushell` first, as inside, the periods are always verbose.

**Comment by grtcdr** [comment]
Take your time, I'll push these changes in the meantime as they make  everything much easier for the end-user.

**Comment by grtcdr** [comment]
v4.0.1 is out!

**Comment by 0323pin** [comment]
@grtcdr All good now, closing this issue.

Is it possible to output time in the format 17:50 instead of 5:50 PM?

Thank you.

**Comment by grtcdr** [comment]
Yeah, that would be possible in a future update, since _formatters_ were designed with this very purpose in mind.

**Comment by 0323pin** [comment]
Yeah, no need for it now 😄 

merged, https://mail-index.netbsd.org/pkgsrc-changes/2021/12/10/msg244770.html

**Comment by 0323pin** [comment]
@grtcdr Sorry for posting on a closed issue but, I didn't think it was worthy starting a new one.
So, I've built and merged 0.5.0 yesterday and while building I saw something along the line, _"...using expat for documentation..."_ 
Can't remember the exact wording but, thought I'd wait and see if bulk-builds are failing with an unmet dependency issue.

Are you using `py-expat` to generate some sort of man page?
